XMonad/Actions/Menu/KeyParse.hs

```diff
@@ -0,0 +1,347 @@
+module XMonad.Actions.Menu.KeyParse where
+
+{-|
+
+Module      : XMonad.Menu.Actions.KeyParse
+Description : A set of tools to run and render interactive menus
+
+Maintainer  : David Janssen <janssen.dhj@gmail.com>
+Stability   : unstable
+Portability : unportable
+
+This code is taken almost entirely from the key-string parsing capabilities of
+EZConfig, written by Brent Yorgey. EZConfig does not export any of these
+functions and I did not want to modify EZConfig itself, so I extracted the code,
```

Comment:
So I can understand your impulse here, but it's still painful to see so much duplicated code.  Let's talk about ideas for reuse.  Modifying EZConfig is certainly on the table as long as we don't change its current API.  Do you have ideas for how we could unify the two?